tower-grpc-interop/src/client.rs

```diff
@@ -241,281 +537,20 @@ impl Testcase {
-                    pb::StreamingOutputCallRequest>();
-
-                sender.unbounded_send(make_req(0)).unwrap();
-
-                core.run(client
-                    .full_duplex_call(Request::new(receiver
-                        .map_err(|_error| panic!("Receiver stream should not error!"))))
-                    .map_err(|tower_error| -> Box<Error> {
-                        Box::new(tower_error)
-                    })
-                    .and_then(|response_stream| {
-                        perform_ping_pong(State {
-                            sender,
-                            stream: Box::new(response_stream.into_inner()),
-                            responses: vec![],
-                            assertions: vec![],
-                        })
-                    })
-                    .map(|(responses, mut assertions)| {
-                        let actual_response_lengths = response_lengths(&responses);
-                        assertions.push(test_assert!(
-                            "there should be four responses",
-                            responses.len() == RESPONSE_LENGTHS.len(),
-                            format!("{:?}={:?}", responses.len(), RESPONSE_LENGTHS.len())
-                        ));
-                        assertions.push(test_assert!(
-                            "the response payload sizes should match input",
-                            RESPONSE_LENGTHS == actual_response_lengths.as_slice(),
-                            format!("{:?}={:?}", RESPONSE_LENGTHS, actual_response_lengths)
-                        ));
-                        assertions
-                    })
-                    .then(&assert_success))
-            },
-            Testcase::empty_stream => {
-                let stream = stream::iter_ok(Vec::<pb::StreamingOutputCallRequest>::new());
-                core.run(client.full_duplex_call(Request::new(stream))
-                    .map_err(|tower_error| -> Box<Error> {
-                        Box::new(tower_error)
-                    })
-                    .and_then(|response_stream| {
-                        // Convert the stream into a plain Vec
-                        response_stream.into_inner()
-                            .collect()
-                            .map_err(|tower_error| -> Box<Error> {
-                                Box::new(tower_error)
-                            })
-                    })
-                    .map(|responses: Vec<pb::StreamingOutputCallResponse>| -> Vec<TestAssertion> {
-                        vec![
-                            test_assert!(
-                                "there should be no responses",
-                                responses.len() == 0,
-                                format!("responses.len()={:?}", responses.len())
-                            ),
-                        ]
-                    })
-                    .then(&assert_success))
-            },
+            Testcase::empty_unary =>
```

Comment:
This is basically how I initially wanted to structure this function originally --- it's _so_ much less ugly than putting all the test code in this method --- but I think something got in the way of that at the time.

Thanks so much for fixing this up! 👍